### DISCUSSION TITLE
About bibliography and annotations

### DISCUSSION THREAD

**Discussion by bepolymathe** [EdoFro/Freeplane_MarkdownHelper#13] [Ideas]
Hi @EdoFro 

On another subject. I don't know if you have seen the information about the existence of [an addon that allows to integrate better freeplane and zotero](https://github.com/petervelosy/freeplane-zotero). 

What seems to me very interesting since version 6 of Zotero is the possibility to annotate PDFs in a very nice way and to get annotations and comments in markdown. Hence the idea that [we started a discussion](https://github.com/petervelosy/freeplane-zotero/issues/7) about importing annotations into freeplane from zotero. 

I don't know if this discussion would be of interest to you but I see a great potential in being able to integrate annotations and biblio references when writing a draft with MarkdownHelper. If you're interested, your ideas would surely be of great help.

**Comment by EdoFro** [comment]
hi @bepolymathe,
I have never used Zotero. Today I googled about it and looked some presentations. I looked at your discussion with  @petervelosy but I didn't  have the time to read it carefully.
I will read it and look if I can think of some changes or add some new features to MDH that can help you.

Let me look at your discussion and understand it better to see if I have questions or ideas.

bye,

edo

**Comment by EdoFro** [comment]
Maybe it can be used https://michel-kraemer.github.io/citeproc-java/ to transform the citation json to a formated node's note using the desired CSL style?

So it can be used by MDH to include it in the markdown document.

**Comment by bepolymathe** [comment]
Hi @EdoFro 

Thank you for your interest in this project ! 

My main point is that MDH is a wonderful tool for presentation drafts or meeting notes (or technical documentation) but is still a bit limited for more complex documents like scientific papers. In particular, it would require the ability to manage bibliographic references and footnotes. But the potential is great. 

The discovery of this freeplane addon changes the game because it allows to use Zotero for bibliography management which is a free software as old and solid as Freeplane. I store at least 5000 bibliographic references in it (+ PDF of articles) for more than 10 years. Since version 6 of the software, it is possible to annotate PDFs and export them in markdown. The annotations are stored separately from the file. Importing the reference, the annotations and then reorganizing everything in Freeplane + writing with MDH would be incredibly powerful for all students, researchers, journalists or others... 

> So it can be used by MDH to include it in the markdown document.

Sounds interesting ! I will look at what I understand about it. I'll also let @petervelosy you react on the technical side.

**Reply by EdoFro** [reply]
> My main point is that MDH is a wonderful tool for presentation drafts or meeting notes (or technical documentation) but is still a bit limited for more complex documents like scientific papers. In particular, it would require the ability to manage bibliographic references and footnotes. But the potential is great.

ok. Here are some ideas. 

What about adding some new MDH nodes:

1. [footnotes](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#footnotes): but I think this works only in some markdown application, like Github. I don't know if the markdown java library Freeplane (and MDH) uses has this capability.
It shows the footnotes in the same page, not at the end of the document (but I really don't know how could it work in MDH, because the exported document has no pages. Surely depends where do you import or paste the document).
I think it may be useful for your citation needs.

3. [Section links](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#section-links): this looks interesting too, but it links only to headings in a one way direction (link to heading)

4. **"MDH Biblio node":** a dedicated **MDH node** that collects all citations and builds a formated Bibliography section in the document  (like the **_TOC node_** that collects all headings).
5. A **"citation Number" node**: that inserts the correlative number that refers to the citation number in the bibliography. (this will need a command to update the numbers from the citations before exporting)

**Reply by EdoFro** [reply]
> > So it can be used by MDH to include it in the markdown document.
> 
> Sounds interesting ! I will look at what I understand about it. I'll also let @petervelosy you react on the technical side.

Maybe Zotero has this API function itself: to get the formated text of a citation using a defined CSL style.

**Reply by EdoFro** [reply]
> 1. [footnotes](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#footnotes): but I think this works only in some markdown application, like Github. I don't know if the markdown java library Freeplane (and MDH) uses has this capability.

It doesn't render ok in Freeplane:

![image](https://user-images.githubusercontent.com/35700575/217539127-611a70c2-2b29-49d5-ab40-ded7f2420a9d.png)

but the exported file can be correctly rendered by other MD applications.

Here is a Notepad++ example using the same text that I used in the Freeplane's note:

![image](https://user-images.githubusercontent.com/35700575/217540442-e63dc08a-575d-454c-9ac3-53ce511b417e.png)

bye,
edo

**Reply by bepolymathe** [reply]
Hi @EdoFro 

Thanks for reply.

> What about adding some new MDH nodes:

Yes ! 

### Footnotes

It's true that it's a pity that the footnotes are not rendered in Freeplane but the fact that they are retrieved in the export is already excellent. I just tried it with Zettlr (which has a good integration with Pandoc) and it is already very useful. 

If we wanted to go further, perhaps MDH should manage the footnotes as it manages the lists 🤔 If This would allow to group them visually and to place them at the end of the document when exporting...

![image](https://user-images.githubusercontent.com/29939805/217555584-ba4c5cc7-6475-4d0b-a01f-3ec4c731fb5d.png)

### Sections links 

It is necessary to think but it could be interesting to make references in the text.

### MDH Biblio node

That would be great

### "citation Number" node

That would be very convenient ! The update of the numbers could be used also for the footnotes 🤔 
Otherwise, we could also rely on a very useful feature of Zotero used by many people: better-bibtex. It allows to export automatically (at each modification of the bibliography in Zotero) a .bib file usable with LaTeX or scientific markdown editors like Zettlr. This export generates a unique bibtex key per reference...

**Comment by bepolymathe** [comment]
Yes, it is possible to retrieve a lot of information via the Zotero API, but if I understand correctly, many recent software allowing markdown editing and document structuring (Zettlr, Logseq, Obsidian, etc.) decide to use the export via [better-bibtex](https://retorque.re/zotero-better-bibtex/) because working on the .bib file is faster than querying the API.
